src/Geopilot.Frontend/src/api/fetch.ts

```diff
@@ -1,15 +1,51 @@
-export const runFetch = async (url: string, contentType?: string) => {
-  const response = await fetch(url, {
-    method: "GET",
-  });
-
-  if (response.ok) {
-    const responseContentType = response.headers.get("content-type");
-    if (responseContentType?.indexOf("application/json") !== -1) {
-      return await response.json();
-    } else if (!contentType || responseContentType?.includes(contentType)) {
-      return await response.text();
+export interface FetchParams {
+  url: string;
+  method?: FetchMethod;
+  contentType?: FetchContentType;
+  body?: BodyInit | null;
+  onSuccess: (response: unknown) => void;
+  onError?: (error: string, status?: number) => void;
+}
+
+export enum FetchMethod {
+  GET = "GET",
+  POST = "POST",
+  PUT = "PUT",
+  DELETE = "DELETE",
+}
+
+export enum FetchContentType {
+  JSON = "application/json",
+  TEXT = "text/plain",
+}
+
+export const runFetch = async ({
```

Comment:
PP: Könnte in einen Context gepackt werden und dann `const api = useApi()` & `api.fetch()` aufgerufen werden. Das gibt uns die Möglichkeit die `Auth` aus dem `AuthContext` zu lesen & so sicher zu stellen, dass wir keine asynchronen Probleme bekommen.